compiler/rustc_typeck/src/check/callee.rs

```diff
@@ -307,6 +307,37 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 
+    /// Give appropriate suggestion when encountering `[("a", 0) ("b", 1)]`, where the
+    /// likely intention is to create an array containing tuples.
+    fn identify_bad_array_def(
+        &self,
+        err: &mut DiagnosticBuilder<'a>,
+        call_expr: &'tcx hir::Expr<'tcx>,
+        callee_node: &hir::ExprKind<'_>,
```

Comment:
nit: it might be slightly cleaner to just pass the `callee_expr` here rather than the individual parts?

Reply:
Done.